examples/medplum-photon-integration/src/bots/create-photon-prescription.ts

```diff
@@ -0,0 +1,298 @@
+  medplum: MedplumClient,
+  medicationCode?: CodeableConcept
+): Promise<string | undefined> {
+  if (!medicationCode) {
+    throw new Error('Medication must have a code');
+  }
+
+  // If possible get the code from the MedicationKnowledge resource with the given code
+  const photonId = await getPhotonIdByCoding(medplum, medicationCode);
+
+  if (photonId) {
+    return photonId;
+  }
+
+  // Check for an RXNorm code and get the display string
+  let medicationName = medicationCode.coding?.find(
+    (code) => code.system === 'http://www.nlm.nih.gov/research/umls/rxnorm'
+  )?.display;
+
+  // If there is no RXNorm code, check for an NDC code and get the display string
+  if (!medicationName) {
+    medicationName = medicationCode.coding?.find((code) => code.system === 'http://hl7.org/fhir/sid/ndc')?.display;
+  }
+
+  // If there is no RXNorm or NDC throw an error
+  if (!medicationName) {
+    throw new Error('Could not find medication in Photon');
+  }
+
+  // Build the query
+  const query = `
+    query treatmentOptions($searchTerm: String!) {
+      treatmentOptions(searchTerm: $searchTerm) {
+        medicationId
+        name
+        ndc
+      }
+    }
+  `;
+
+  const variables = { searchTerm: medicationName };
+
+  const body = JSON.stringify({ query, variables });
+
+  try {
+    // Query Photon's clinical API
+    const response = await fetch('https://clinical-api.neutron.health/graphql', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'x-photon-auth-token-type': 'auth0',
+        'x-photon-auth-token': authToken,
+      },
+      body,
+    });
+
+    if (!response.ok) {
+      throw new Error(`HTTP Error! Status: ${response.status} ${response.statusText}`);
+    }
+
+    const result = await response.json();
+    // Return the medication ID
+    return result.data.treatmentOptions?.[0].medicationId;
+  } catch (err) {
+    throw new Error(normalizeErrorString(err));
+  }
+}
+
+/**
+ * Searches for Medication Knowledge resources with the given code, then checks if the Photon ID for that treatment is stored on
+ * them. If it is, it returns the Photon Treatment ID.
+ *
+ * @param medplum - Medplum Client to search for MedicationKnowledge resources
+ * @param medicationCode - The codeable concept of the medication for the prescription
+ * @returns - The Photon ID if it is store on the MedicationKnowledge
+ */
+export async function getPhotonIdByCoding(
+  medplum: MedplumClient,
+  medicationCode: CodeableConcept
+): Promise<string | undefined> {
+  // Get the medication's code by RXNorm or NDC
+  let code =
+    getCodeBySystem(medicationCode, 'http://www.nlm.nih.gov/research/umls/rxnorm') ??
+    getCodeBySystem(medicationCode, 'http://hl7.org/fhir/sid/ndc');
+
+  // If there is a code, search for the relevant MedicationKnowledge in your project
+  if (code) {
+    const medicationKnowledge = await medplum.searchOne('MedicationKnowledge', {
+      code,
+    });
```

Comment:
My tests for this function are failing when I get to this point. They are being passed the correct code, but the search is returning undefined.